src/main/java/io/confluent/kafkarest/ConsumerManager.java

```diff
@@ -160,49 +163,57 @@ public String createConsumer(String group, ConsumerInstanceConfig instanceConfig
       // To support the old consumer interface with broken peek()/missing poll(timeout)
       // functionality, we always use a timeout. This can't perfectly guarantee a total request
       // timeout, but can get as close as this timeout's value
-      props.setProperty("consumer.timeout.ms", ((Integer) iteratorTimeoutMs).toString());
+//      props.setProperty("consumer.timeout.ms", ((Integer) iteratorTimeoutMs).toString());
       if (instanceConfig.getAutoCommitEnable() != null) {
-        props.setProperty("auto.commit.enable", instanceConfig.getAutoCommitEnable());
+        props.setProperty("enable.auto.commit", instanceConfig.getAutoCommitEnable());
       } else {
-        props.setProperty("auto.commit.enable", "false");
+        props.setProperty("enable.auto.commit", "false");
       }
       if (instanceConfig.getAutoOffsetReset() != null) {
         props.setProperty("auto.offset.reset", instanceConfig.getAutoOffsetReset());
       }
-      ConsumerConnector consumer;
+
       try {
-        if (consumerFactory == null) {
-          consumer = Consumer.createJavaConsumerConnector(new ConsumerConfig(props));
+        ConsumerFactory factory;
+        if (consumerFactory != null) {
+          factory = consumerFactory;
         } else {
-          consumer = consumerFactory.createConsumer(new ConsumerConfig(props));
+          factory = new ConsumerFactory() {
```

Comment:
Moved initialization to constructor. https://github.com/confluentinc/kafka-rest/pull/216/commits/b8a2cf58273d8038ffe30a630f52d1490b163002
Thanks.